Gutenberg: Update Publicize to use inline icons

tyxla opened the pull request:
#### Changes proposed in this Pull Request

* Introduce a new `PublicizeServiceIcon` component with inline icons for Publicize.
* Use that component to render the same icons in any environment.

#### Testing instructions

* Checkout this branch in Calypso.
* Load http://calypso.localhost:3000/block-editor/post/:site where `:site` is a WP.com simple site.
* Click the Jetpack icon in the toolbar.
* Connect one of each Publicize services.
* Verify icons look well.
* Start a JN site with this branch - `gutenpack-jn`.
* Connect the site and activate recommended features.
* Start a new post.
* Click the Jetpack icon in the toolbar, or the Publish button.
* Connect one of each Publicize services.
* Verify icons look well.

Fixes #28953

simison commented on client/gutenberg/extensions/publicize/editor.scss:
```diff
@@ -24,12 +24,8 @@
 
 .jetpack-publicize-gutenberg-social-icon {
-	font-size: 2em;
+	fill: #555d66;
```
Should we use a Gutenberg color variable here to make migrating later on easier?

https://github.com/WordPress/gutenberg/blob/0b30c73a19f6ccb73e8d8e4a43819fc1075080b9/assets/stylesheets/_colors.scss#L11

simison commented on client/gutenberg/extensions/publicize/service-icon.jsx:
```diff
@@ -0,0 +1,73 @@
+ */
+// @TODO: Import those from https://github.com/Automattic/social-logos when that's possible
+const FacebookIcon = (
+	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
>unless Icon component already does that for us?

Confirming that it does indeed add accessibility attributes already:

<img width="1201" alt="image" src="https://user-images.githubusercontent.com/87168/49818677-06ffce80-fd7c-11e8-803d-6e271629445c.png">

sirreal replied:
Shouldn't we be using all the Gutenberg provided SVG primitives? `Circle`, `G`, `Rect`, `Path`, `Polygon`, and `SVG`.

https://github.com/WordPress/gutenberg/blob/0416bae17c52b0a11ec1075c0928f879264b7d75/packages/components/src/primitives/svg/index.js#L6-L12

tyxla replied:
We could, I'll explore.

tyxla replied:
@sirreal there it goes: https://github.com/Automattic/wp-calypso/pull/29348

simison commented on client/gutenberg/extensions/publicize/service-icon.jsx:
```diff
@@ -0,0 +1,73 @@
+/**
+ * Module variables
+ */
+// @TODO: Import those from https://github.com/Automattic/social-logos when that's possible
```
This could maybe provide little more context for future lookers.

i.e. why it isn't possible now?

tyxla replied:
Sure 👍 https://github.com/Automattic/wp-calypso/pull/29317/commits/7d7cabab99d976360407891380d627bcdb1c5e0d

ockham commented:
Looks like we're about to remove Google+ from Publicize very soon (p58i-7iC-p2 #comment-40654) :upside_down_face: So should we just use Gutenberg's SVGs after all? https://github.com/Automattic/wp-calypso/issues/28953#issuecomment-444019714 (To avoid some duplication)

MichaelArestad commented:
Can we use this Twitter icon instead?

https://github.com/Automattic/social-logos/blob/master/svg-min/twitter-alt.svg

MichaelArestad commented:
It would also be good to add their brand colors to the icon.

alisterscott commented:
Tests well for me

<img width="281" alt="screen shot 2018-12-12 at 4 52 52 pm" src="https://user-images.githubusercontent.com/128826/49851948-66b6b200-fe2e-11e8-9501-56339127cd35.png">

tyxla commented on client/gutenberg/extensions/publicize/editor.scss:
```diff
@@ -24,12 +24,8 @@
 .jetpack-publicize-gutenberg-social-icon {
-	font-size: 2em;
+	fill: #555d66;
 	margin-right: 0.2em;
```
I'll follow up in another PR.

tyxla replied:
@simison there it goes: https://github.com/Automattic/wp-calypso/pull/29343

tyxla commented:
> Looks like we're about to remove Google+ from Publicize very soon (p58i-7iC-p2 #comment-40654) :upside_down_face: So should we just use Gutenberg's SVGs after all? https://github.com/Automattic/wp-calypso/issues/28953#issuecomment-444019714 (To avoid some duplication)

Well, we still can't - Gutenberg doesn't supply a LinkedIn icon, and it doesn't directly expose the Tumblr icon. So we're stuck with ours for now :)

tyxla commented:
I'll also follow up on the design feedback on subsequent PRs.

tyxla commented:
@MichaelArestad I've addressed your feedback in additional PRs:

> Can we use this Twitter icon instead?
>
> https://github.com/Automattic/social-logos/blob/master/svg-min/twitter-alt.svg

Addressed in https://github.com/Automattic/wp-calypso/pull/29341

> It would also be good to add their brand colors to the icon.

Addressed in https://github.com/Automattic/wp-calypso/pull/29342

Those 2 PRs are independent and iteratively improve the icons. I'd appreciate your feedback when you get a chance.
